src/example_robot_framework/turtlesim_ros2.py

```diff
@@ -0,0 +1,111 @@
+import rclpy
+import threading
+from rclpy.node import Node
+from rclpy.action import ActionClient
+from nav_msgs.msg import Odometry
+from geometry_msgs.msg import Twist
+from turtlesim.msg import Pose
+from std_srvs.srv import Empty
+from example_robot_framework import SelfSpinningNode
+from turtlesim.action import RotateAbsolute
+
+
+
+class TurtlesimExampleNode(SelfSpinningNode):
+    def __init__(self):
+        super().__init__('robot_framework_turtlesim_node')
+        self.__movement_detected = False
+        self.__expecting_to_move = False
+        self.action_result = None
+        self.action_result_condition = threading.Condition()
+
+    def publish_velocity(self, vel):
+        move_cmd = Twist()
+        move_cmd.linear.x = vel
+        self.cmd_vel_publisher.publish(move_cmd)
+        self.__movement_detected = False
+        self.__expecting_to_move = True
+
+    def cmd_vel_listener_callback(self, pose_msg):
+        print("pose received")
+        if self.__expecting_to_move is False:
+            return
+        self.__movement_detected = True
+
+    def has_moved(self):
+        return self.__movement_detected
+
+    def reset_turtle_service(self, timeout=1.0):
+        print('reset turtle')
+        try:
+            service_request = Empty.Request()
+            while not self.reset_turtle_client.wait_for_service(timeout_sec=timeout):
+                print('service not available, waiting...')
+            self.reset_turtle_client.call(service_request)
+            return True
+        except:
+            return False
+
+    def reset_turtle_service_async(self, timeout=1.0):
+        print('resetting turtle')
+        try:
+            service_request = Empty.Request()
+            while not self.reset_turtle_client.wait_for_service(timeout_sec=timeout):
+                print('service not available, waiting...')
```

Comment:
shall this be logged instead of printed?

Reply:
It can be both logged or printed. The reason I am printing it instead is because it will appear in the "report/log" which you get after a test is completed